File: UICPC/25/g.cpp
#include <bits/stdc++.h>
#include <string.h>
using namespace std;
int RANGE;
void countSort(string arr)
{
    char output[strlen(arr)];

    int count[RANGE + 1], i;
    memset(count, 0, sizeof(count));
    for (i = 0; arr[i]; ++i)
        ++count[arr[i]];
    for (i = 1; i <= RANGE; ++i)
        count[i] += count[i - 1];
    for (i = 0; arr[i]; ++i) {
        output[count[arr[i]] - 1] = arr[i];
        --count[arr[i]];
    }
    for (i = 0; arr[i]; ++i)
        arr[i] = output[i];
}
 
int main()
{
    cin>>RANGE;
    string st;
    cin>>st;
    /**
    char arr[n];
    for (int i = 0; i < n; i++)
    {
        arr[i]=st[i];
    }**/
    countSort(st);
    cout << arr;
    return 0;
}

File: UICPC/25/g.py
def countSort(arr):
    output = [0 for i in range(len(arr))]
    count = [0 for i in range(256)]
    ans = ["" for _ in arr]
    for i in arr:
        count[ord(i)] += 1
    for i in range(256):
        count[i] += count[i-1]
    for i in range(len(arr)):
        output[count[ord(arr[i])]-1] = arr[i]
        count[ord(arr[i])] -= 1
    for i in range(len(arr)):
        ans[i] = output[i]
    return ans
n = int(input())
arr = list(input())
ans = countSort(arr)
print(ans)